habitat_baselines/rl/ppo/ppo_trainer.py

```diff
@@ -0,0 +1,519 @@
+
+                # checkpoint model
+                if update % ppo_cfg.checkpoint_interval == 0:
+                    self.save_checkpoint(f"ckpt.{count_checkpoints}.pth")
+                    count_checkpoints += 1
+
+    def eval(self):
+        ppo_cfg = self.config.TRAINER.RL.PPO
+
+        assert (len(ppo_cfg.model_path) > 0) != (
+            len(ppo_cfg.tracking_model_dir) > 0
+        ), "Must specify a single model or a directory of models, but not both"
+        if "tensorboard" in ppo_cfg.video_option:
+            assert (
+                ppo_cfg.tensorboard_dir is not None
+            ), "Must specify a tensorboard directory for video display"
+        if "disk" in ppo_cfg.video_option:
+            assert (
+                ppo_cfg.video_dir is not None
+            ), "Must specify a directory for storing videos on disk"
+
+        with get_tensorboard_writer(
+            ppo_cfg.tensorboard_dir, purge_step=0, flush_secs=30
+        ) as writer:
+            if len(ppo_cfg.model_path) > 0:
+                # evaluate singe checkpoint
+                self._eval_checkpoint(ppo_cfg.model_path, ppo_cfg, writer)
+            else:
+                # evaluate multiple checkpoints in order
+                prev_ckpt_ind = -1
+                while True:
+                    current_ckpt = None
+                    while current_ckpt is None:
+                        current_ckpt = poll_checkpoint_folder(
+                            ppo_cfg.tracking_model_dir, prev_ckpt_ind
+                        )
+                        time.sleep(2)  # sleep for 2 secs before polling again
+                    logger.warning(
+                        "=============current_ckpt: {}=============".format(
+                            current_ckpt
+                        )
+                    )
+                    prev_ckpt_ind += 1
+                    self._eval_checkpoint(
+                        current_ckpt,
+                        ppo_cfg,
+                        writer,
+                        cur_ckpt_idx=prev_ckpt_ind,
+                    )
+
+    def _eval_checkpoint(self, checkpoint_path, args, writer, cur_ckpt_idx=0):
+        device = torch.device("cuda", args.pth_gpu_id)
+        config = self.config.clone()
+        config.defrost()
+        config.DATASET.SPLIT = "val"
+        agent_sensors = args.sensors.strip().split(",")
+        for sensor in agent_sensors:
+            assert sensor in ["RGB_SENSOR", "DEPTH_SENSOR"]
+        config.SIMULATOR.AGENT_0.SENSORS = agent_sensors
+        if args.video_option:
+            config.TASK.MEASUREMENTS.append("TOP_DOWN_MAP")
+            config.TASK.MEASUREMENTS.append("COLLISIONS")
+        config.freeze()
+
+        logger.info(f"env config: {config}")
+        self.envs = construct_envs(config, NavRLEnv)
+
+        ckpt = torch.load(checkpoint_path, map_location=device)
+
+        actor_critic = Policy(
```

Comment:
Use `self._setup` here?

Reply:
Thanks for spotting that. I'll actually load saved config and merge them with config specified on evaluation.